frontend/src/utils/storageUtils.ts

```diff
@@ -85,9 +84,7 @@ export const initChainsState = () => {
 	// reset chains if anything is wrong or missing from localStorage values
 	if(!chain || 
 		!destinationChain || 
-		chain === destinationChain ||
-		!getSrcChains().some(x => x === chain) ||
-		!getDstChains(chain).some(x => x === destinationChain)
+		chain === destinationChain
```

Comment:
should'nt we check somehow if `(src)chain` and `destinationChain` are inside `allowedDirections`?